### PR TITLE
Fix empty request / response bodies

### LINKED ISSUE #1109: Responses without bodies also omit headers from generated output.
**Description**

When an OpenAPI spec does not specify a response body for a response (say `204 No Content`), the generated type for that response becomes `never`, which omits any headers that might be specfied for that response.

| Name                 | Version                      |
|:---------------------|:-----------------------------|
| `openapi-typescript` | `6.2.4`                      |
| Node.js              | `20.1.0`                      |
| OS + version         | `macOS 13.3.1` |

**Reproduction**

Convert the following OpenAPI schema to TypeScript using `openapi-typescript`:

```YAML
openapi: "3.0.2"
info:
  title: Response header bug example
  version: "1.0.0"
  contact:
    email: d.leinhaeuser@reply.de
servers:
  - description: local server
    url: http://localhost
paths:
  "/foo":
    get:
      responses:
        "204":
          description: No Content
          headers:
            foo:
              schema:
                type: string
  "/bar":
    get:
      responses:
        "200":
          description: OK
          headers:
            "bar":
              schema:
                type: string
          content:
            "application/json":
              schema:
                type: object
```

**Expected result**

`paths["/foo"]["get"]["responses"][204]` is `{ headers: { foo?: string; } }; content: {}; }` or something similar instead of `never`.

**Checklist**

- [ ] I’m willing to open a PR (see [CONTRIBUTING.md](https://github.com/drwpow/openapi-typescript/blob/main/CONTRIBUTING.md))

### PR DESCRIPTION
**Pull request by pvanagtmaal**
## Changes

Fixes #1109 and #1174

## How to Review

Nothing special

## Checklist

- [x] Unit tests updated
- [x] `docs/` updated (if necessary)
- [x] `pnpm run update:examples` run (only applicable for openapi-typescript)

### CODE REVIEW

**Comment by pvanagtmaal** [comment]
@drwpow The tests that are failing are because we cannot simply return `never` for responses anymore because the headers could still be set. Do you consider this a problem / breaking change? IMO it is not a problem, since types of response objects that were `never` wouldn't be used anyways. 🤔

**Comment by pvanagtmaal** [comment]
Not sure why the CI is failing, tests are passing locally:
```
~/openapi-typescript/packages/openapi-typescript on  fix/empy-response-or-request-objects [$?] is 📦 v6.4.3 via  v18.10.0
❯ pnpm run test

> openapi-typescript@6.4.3 test ~/openapi-typescript/packages/openapi-typescript
> run-p -s test:*


 RUN  v0.33.0 ~/openapi-typescript/packages/openapi-typescript

 ✓ test/cli.test.ts (8) 22314ms
 ✓ test/cjs.test.ts (1) 709ms
 ✓ test/yaml.test.ts (1) 430ms
 ✓ test/utils.test.ts (35)
 ✓ test/schema-object.test.ts (67)
 ✓ test/load.test.ts (7)
 ✓ test/response-object.test.ts (2)
 ✓ test/operation-object.test.ts (1)
 ✓ test/index.test.ts (26) 22131ms
 ✓ test/paths-object.test.ts (3)
 ✓ test/components-object.test.ts (5)
 ✓ test/request-body-object.test.ts (2)
 ✓ test/invalid.test.ts (2)
 ✓ test/path-item-object.test.ts (3)
 ✓ test/webhooks-object.test.ts (2)
 ✓ test/header-object.test.ts (1)

 Test Files  16 passed (16)
      Tests  166 passed (166)
   Start at  10:02:05
   Duration  22.87s (transform 465ms, setup 0ms, collect 2.46s, tests 45.70s, environment 3ms, prepare 1.76s)
   ```

**Comment by drwpow** [comment]
@pvanagtmaal you’ll need to update the snapshots in `examples/*` with the newly-generated types. The tests are failing on my machine. Also be sure to run `pnpm run build` so you’re testing the freshly-built client (many tests test the actual compiled CLI version just to make sure it’s working, and that won’t rebuild automatically just from editing `.ts` files unless you’re running `pnpm run dev`)!

File: packages/openapi-typescript/examples/digital-ocean-api.ts [hunk 1]
```diff
@@ -16033,7 +16041,14 @@ export interface external {
     page?: number;
     region?: external["shared/attributes/region_slug.yml"];
   };
-  "shared/responses/accepted.yml": never
+  "shared/responses/accepted.yml": {
+    headers: {
```
Comment (drwpow) [inline packages/openapi-typescript/examples/digital-ocean-api.ts:16045]: 🎉 🎉 🎉 

These were the type of test failures we like to see 😄

**Review by drwpow — APPROVED** [review]
Thank you! This is a huge fix 🙏

**Pull request merged**